sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/webapp/AadOAuth2UserService.java

```diff
@@ -83,46 +75,77 @@ public AadOAuth2UserService(AadAuthenticationProperties properties, RestTemplate
+    }
+
+    private AadOAuth2UserService(AadAuthenticationProperties properties,
+                                 GraphClient graphClient) {
         allowedGroupNames = Optional.ofNullable(properties)
-                                    .map(AadAuthenticationProperties::getUserGroup)
-                                    .map(AadAuthenticationProperties.UserGroupProperties::getAllowedGroupNames)
-                                    .orElseGet(Collections::emptyList);
+            .map(AadAuthenticationProperties::getUserGroup)
+            .map(AadAuthenticationProperties.UserGroupProperties::getAllowedGroupNames)
+            .orElseGet(Collections::emptyList);
         allowedGroupIds = Optional.ofNullable(properties)
-                                  .map(AadAuthenticationProperties::getUserGroup)
-                                  .map(AadAuthenticationProperties.UserGroupProperties::getAllowedGroupIds)
-                                  .orElseGet(Collections::emptySet);
-        DefaultOAuth2UserService oAuth2UserService = new DefaultOAuth2UserService();
-        oAuth2UserService.setRestOperations(createOAuth2ErrorResponseHandledRestTemplate(restTemplateBuilder));
-        this.oidcUserService = new OidcUserService();
-        this.oidcUserService.setOauth2UserService(oAuth2UserService);
+            .map(AadAuthenticationProperties::getUserGroup)
+            .map(AadAuthenticationProperties.UserGroupProperties::getAllowedGroupIds)
+            .orElseGet(Collections::emptySet);
         this.graphClient = graphClient;
     }
 
     /**
-     * Returns an {@link OAuth2User} after obtaining the user attributes of the End-User
-     * from the UserInfo Endpoint.
+     * Returns a {@link DefaultOidcUser} instance.
+     * <p/>
+     *
+     * The {@link DefaultOidcUser} instance is constructed with {@link GrantedAuthority}, {@link OidcIdToken} and nameAttributeKey.
+     * <a href="https://learn.microsoft.com/azure/active-directory/develop/userinfo#consider-using-an-id-token-instead">Azure AD</a> suggests get userinfo from idToken instead from the UserInfo Endpoint,
+     * this implementation will not get userinfo from the UserInfo Endpoint. Calling {@link org.springframework.security.oauth2.core.oidc.user.OidcUser#getUserInfo()} with the return instance will return null.
+     *
+     * <p/>
      *
      * @param userRequest the user request
-     * @return an {@link OAuth2User}
-     * @throws OAuth2AuthenticationException if an error occurs while attempting to obtain
-     * the user attributes from the UserInfo Endpoint
+     *
+     * @return a {@link DefaultOidcUser} instance.
+     *
+     * @throws OAuth2AuthenticationException if an error occurs.
      */
     @Override
     public OidcUser loadUser(OidcUserRequest userRequest) throws OAuth2AuthenticationException {
-        // Delegate to the default implementation for loading a user
-        OidcUser oidcUser = oidcUserService.loadUser(userRequest);
```

Comment:
This service is an extension for `OidcUserService`, if we remove the dependency of oidcUserService, it will not be a full Open ID Connect process that Spring Security implements. Some `OAuth2AuthenticationException`s will be ignored the new implementation will not throw `OAuth2AuthenticationException `.

Reply:
> This service is an extension for `OidcUserService`

I think it's an implementation of OAuth2UserService.

> if we remove the dependency of oidcUserService, it will not be a full Open ID Connect process that Spring Security implements. 

Yes, there are differences between our implementation Spring Security .
What dou mean the  `Open ID Connect process`?

> Some OAuth2AuthenticationExceptions will be ignored the new implementation will not throw OAuth2AuthenticationException .

So, is there any problem?